### PR TITLE
[improve][broker][PIP-149]make setPersistence method async in Namespaces

### PR DESCRIPTION
**Pull request by Pomelongan**
### Motivation
See https://github.com/apache/pulsar/issues/14365

### Verifying this change

- [x] Make sure that the change passes the CI checks.
This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 
  
- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -1559,27 +1559,17 @@ protected CompletableFuture<Void> internalDeletePersistenceAsync() {
...
+        return validateNamespacePolicyOperationAsync(namespaceName, PolicyName.PERSISTENCE, PolicyOperation.WRITE)
+                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync())
+                .thenAccept(__ -> validatePersistencePolicies(persistence))
+                .thenCompose(__ -> {
+                    return CompletableFuture.supplyAsync(() -> {
+                        validatePersistencePolicies(persistence);
+                        return null;
+                    });
+                })
+                .thenCompose(__ -> doUpdatePersistenceAsync(null));
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1572]: `doUpdatePersistenceAsync(null)` -> `doUpdatePersistenceAsync(persistence)`

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -1559,27 +1559,17 @@ protected CompletableFuture<Void> internalDeletePersistenceAsync() {
...
+    protected CompletableFuture<Void> internalSetPersistenceAsync(PersistencePolicies persistence) {
+        return validateNamespacePolicyOperationAsync(namespaceName, PolicyName.PERSISTENCE, PolicyOperation.WRITE)
+                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync())
+                .thenAccept(__ -> validatePersistencePolicies(persistence))
+                .thenCompose(__ -> {
+                    return CompletableFuture.supplyAsync(() -> {
+                        validatePersistencePolicies(persistence);
+                        return null;
+                    });
+                })
```
Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1571]: We should remove this cold block I think

Comment (AnonHxy) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1571]: I think we should remove this code block from line1566 to line1571

Comment (Pomelongan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: @AnonHxy PTAL

**Review by Technoboy- — APPROVED** [review]

**Comment by AnonHxy** [comment]
Please fix the CI  `org.apache.pulsar.broker.admin.NamespacesTest#testPersistenceUnauthorized`  @Pomelongan

**Comment by Technoboy-** [comment]
Please fix the check style issue.

**Review by AnonHxy — APPROVED** [review]

**Pull request merged**